src/bin/cargo/commands/install.rs

```diff
@@ -172,6 +172,14 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
     let mut from_cwd = false;
 
     let source = if let Some(url) = args.get_one::<String>("git") {
+        if url.contains('#') {
+            return Err(anyhow::format_err!(
+                "invalid git url to install from\n\n\
+                    help: use `--rev <SHA>` to specify a commit"
+            )
+            .into());
+        }
+
```

Comment:
Is this definitively a bad URL or should this be a hint on the existing error?

Reply:
I think it's definitely a bad URL, yes.
From the github page:
![2025-05-05-223239_hyprshot](https://github.com/user-attachments/assets/0b78cb0d-86fa-4b86-b68f-a779c72afcc0)

I can't think of a valid git url with a `#` besides the rev.

Reply:
Cargo supports more than Github.  The question would be more about the git protocol and what is expected around it as we could be turning a success case into an error case.

Reply:
From what I'm seeing [here](https://git-scm.com/docs/http-protocol), other git servers should be the same.